core/templates/hashfuncs.h

```diff
@@ -167,11 +233,24 @@ template <class T>
 class Ref;
 
 struct HashMapHasherDefault {
+	// Generic hash function for any type.
+	template <class T>
+	static _FORCE_INLINE_ uint32_t hash(const T *p_pointer) { return hash_one_uint64((uint64_t)p_pointer); }
+
+	template <class T>
+	static _FORCE_INLINE_ uint32_t hash(const Ref<T> &p_ref) { return hash_one_uint64((uint64_t)p_ref.operator->()); }
+
 	static _FORCE_INLINE_ uint32_t hash(const String &p_string) { return p_string.hash(); }
 	static _FORCE_INLINE_ uint32_t hash(const char *p_cstr) { return hash_djb2(p_cstr); }
-	static _FORCE_INLINE_ uint32_t hash(const uint64_t p_int) { return hash_one_uint64(p_int); }
+	static _FORCE_INLINE_ uint32_t hash(const wchar_t p_wchar) { return (uint32_t)p_wchar; }
```

Comment:
I think this and the ones below likely should be hashed too.

Reply:
For the record they were not hashed before, they were just moved here. They could likely be hashed somehow but since they're ints that fit in `uint32_t` that might not be needed?

Reply:
Yeah but this is actually the issue that we are trying to fix. For sequential indices, if not hashed, robin hood will not work properly because it relies on random gaps on the hash table.

Reply:
I replaced it with the murmurhash3 finalizer function `fmix32()` which does bitmixing pretty well for hashing (although just for the char types, should I use this approach on the other integer types too?)